src/components/main/Main.js

```diff
@@ -31,25 +31,41 @@ function Main() {
     setCards([...cards, newCard]);
   };
 
+  const updateCard = (updatedCard) => {
+    const newCards = cards.map((card) => {
+      if (card.id === updatedCard.id) {
+        return updatedCard;
+      }
+      return card;
+    });
+    setCards([...newCards]);
```

Comment:
The map() method creates a new array, so why are you recreating it here?
And in your case you're applying a map ID comparison to each element of the array, maybe there's a better way to do that?

Reply:
fixed